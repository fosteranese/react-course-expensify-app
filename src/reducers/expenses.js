const expenseReducerDefaultState = [];

export default (state = expenseReducerDefaultState, action) => {
    switch(action.type){
        case 'ADD_EXPENSE':
            return [...state, action.expense];
        case 'EDIT_EXPENSE':
            return state.map((expense) => (expense.id === action.id ? {...expense, ...action.updates} : expense));
        case 'REMOVE_EXPENSE':
            return state.filter(({id})=> (action.id !== id));
        case 'SET_EXPENSES':
            return action.expenses;
        default:
            return state;
    }
};