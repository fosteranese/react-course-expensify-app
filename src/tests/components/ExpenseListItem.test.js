import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';

test(
    'should generate an expense list item component',
    () => {
        const wrapper = new shallow(<ExpenseListItem {...expenses[0]} />);
        expect(wrapper).toMatchSnapshot();
    }
);