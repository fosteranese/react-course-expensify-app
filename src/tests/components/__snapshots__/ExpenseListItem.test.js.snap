// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate an expense list item component 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    $0.19
    January 1st, 1970
  </p>
</div>
`;
